src/components/Dag/index.js

```diff
@@ -88,13 +88,17 @@ function Dag({ id = 'dag-viewer', children } = {}) {
       left: 0,
     },
   });
+  let isUnmounted = false;
   useLifecycles(
     async () => {
       await loadDagJs();
       const Player = window.ConfluxDagPlayer;
       const container = document.getElementById(id);
       const initialSubChains = await fetchDagData();
 
+      if (isUnmounted) {
```

Comment:
This function (the first argument of `useLifecycles`) will only run once when react is mounting the component. I don't say why we should add this check here.

Reply:
component unmount when js is loading or data is loading.(page change)

Reply:
I get your point. It seems that this is [an antipattern](https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html).

The `unmounted` flag may solve the error. But it covers up the underlying problem of dag and scan:

1. dag itself should check the existence of its container
2. scan's `sendRequest` api should use [something](https://github.com/vercel/swr#conditional-fetching) support "cancel a request", it seems [`superagent` don't support it well](https://stackoverflow.com/questions/44119175/reactjs-interrupting-a-post-request-in-superagent)

Reply:
https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html
says call isMounted is anti pattern。because isMounted was once an deprecated method in react component.

I am using **isUnmounted** flag , which name and meaning are totally different. Also i agree dag should check the existing of  dom .

Reply:
> Checking isMounted before calling setState() does eliminate the warning, but it also defeats the purpose of the warning, since now you will never get the warning (even when you should!)
>
> Other uses of isMounted() are similarly erroneous; using isMounted() is a code smell because the only reason you would check is because you think you might be holding a reference after the component has unmounted.

The reason why `isMounted` is an antipattern is explained clearly by the author. The reason remains to sound no matter the `isMounted` is changed to `isUnmounted` or `if mounted` or `if unmounted`.

I recommend using `document.getElementById` to get the container after those `await` and then check against the existence of the container element before create the `player`.

Also, we need to add a try-catch around the `new Player`.

Reply:
move the position of document.getElementById may get around this issue. But add try catch would make sentry not reporting error when some thing throws.

Reply:
> unfortunately we have no callbacks here

We do have callbacks here (any async function has callback), it's just `superagent` don't support canceling request well. We should follow the recommended way to cancel the request.

> But add try catch would make sentry not reporting error when some thing throws.
I think dag.js should add some dom checking, because even if i add document.getElementById outside, it still some times have errors when page jumps.

Say we do add checking for the existence of container in dag. Then dag will throw an error saying `the container should be a dom element`, we still need to catch that error, right? This has nothing to do with sentry. There are unexpected errors and expected errors. We only need to send those unexpected errors to sentry.

The truth here is that dag already checks if the container is a dom element. It just doesn't check if the container exists.

Reply:
cancel the request I think is an optimzation . Because there are still error in page switching. I have already add dom existing check and it's not working well either, i think this is not related to try catch, because error are still throwing duing the player init time. May be change some code in the dag.js would solve the error in the picture i gave.

Reply:
> Cancel the request I think is an optimzation .
Because there are still error in page switching. 

> I think this is not related to try-catch, because error are still throwing during the player init time.

I don't say any causal relationship in these two statements. Anyway, seems we all agree that this needs further investigation in dag.js and it shouldn't be easily resolved by adding an unmounted check. I just added a new issue regarding this #185 

I recommend we just remove the first commit related to dag and ship another commit to dev.